Assets/HoloToolkit/SpatialMapping/Scripts/SpatialMappingSource.cs

```diff
@@ -42,24 +58,41 @@ public struct SurfaceObject
         /// </summary>
         protected virtual Material RenderMaterial { get { return SpatialMappingManager.Instance.SurfaceMaterial; } }
 
+        private readonly List<SurfaceObject> surfaceObjectsWriteable;
+        private readonly ReadOnlyCollection<SurfaceObject> surfaceObjects;
+
+        public SpatialMappingSource()
+        {
+            surfaceObjectsWriteable = new List<SurfaceObject>();
+            surfaceObjects = new ReadOnlyCollection<SurfaceObject>(surfaceObjectsWriteable);
+        }
+
         protected virtual void Awake()
         {
-            SurfaceObjects = new List<SurfaceObject>();
+            // Nothing.
```

Comment:
Since it's protected virtual, I figure there may be child classes overriding it, so I'd leave it in rather than make the breaking change, but I'm open.  Other votes?